server/src/repositories/test_repository.ts

```diff
@@ -0,0 +1,30 @@
+import { inject, injectable } from "inversify";
+import { Pool } from "pg";
+import { TestResult } from "../entities/test_result";
+
+@injectable()
+export class TestRepository {
+    constructor(@inject("DBConnectionPool") private readonly pool: Pool) {}
+
+    async insertTestResult(testResults: TestResult) {
+        const client = await this.pool.connect();
+
+        const queryString = `INSERT INTO test_results 
+                            (patient_id, 
+                             result, 
+                             test_type, 
+                             test_date,
+                             address_id)
+                             VALUES ($1, $2, $3, $4, $5)`;
+
+        client
+            .query(queryString, [
+                testResults.patientId,
+                testResults.result,
+                testResults.testType,
+                testResults.testDate,
```

Comment:
I think you want to explicitly convert the date object here as well

```suggestion
                testResults.testDate.toISOString(),
```